pintc/src/expr.rs

```diff
@@ -196,6 +199,10 @@ pub enum Immediate {
         value: Option<Box<Immediate>>,
         decl: UnionKey,
     },
+    Optional {
+        value: Option<Box<Immediate>>,
+        value_size: usize,
+    },
 }
```

Comment:
Is there a specific reason for adding an `Expr::Optional` _and_ a `Immediate::Optional`?  `Expr::Optional` isn't enough alone?

Reply:
Remove `Expr::Optional` altogether in favour `Expr::Nil`. That's because I also removed `val` so there's only one possibility which is `nil`.